### PR TITLE
#170947668 View All My Request

### PR DESCRIPTION
**Pull request by Kagorora**
#### What does this PR do?

- This PR allow requestors to view their request history

#### Description of Task to be completed?

- create an endpoint to display all the request made by a specific person

- Test `view all my request` endpoint

#### How should this be manually tested?
- run `git clone https://github.com/andela/knights-bn-backend.git` in your terminal
- run `cd knights-bn-backend` in your terminal
- run `git checkout ft-getAl-myRequest-170947668` to go on the branch
- run `npm install` to install all packages. 
- create `.env` file and populate it, refer to `env.example` to see what fill in .env
- create `database` for both `development` and `test`
- fill the names of the databases you created under `.env`
-  run `npm run createTables` to create tables
- run `npm run start` to start the server.
- use `postman` to open `login`, POST `http://localhost:4000/api/v1/auth/login`.
-  Important: Please login using this credentials otherwise due to the absence of the endpoint of 
    making  a request you will have no request available

        `{
                  "email": "willishimw@gmail.com",
                   "password": "Niyonkuru@1"
          }`

- user `GET` then use the following URL `localhost:4000/api/v1/trips/myRequest` to view all the request that was done by `user: willishimw@gmail.com`

#### Any background context you want to provide?

- N/A

#### What are the relevant pivotal tracker stories?

- [#170947668](https://www.pivotaltracker.com/story/show/170947668)

#### Screenshots

      When a user has request history

Screen Shot 2020-02-18 at 18 19 09

     When a user has no request history

Screen Shot 2020-02-18 at 19 29 05

### CODE REVIEW

**Comment by Niyonsengaeric** [comment]
Screen Shot 2020-02-19 at 15 10 47

@Kagorora I followed your PR description but, it stacks on creating the tables!

**Comment by Kagorora** [comment]
> Screen Shot 2020-02-19 at 15 10 47
> 
> @Kagorora I followed your PR description but, it stacks on creating the tables!

@Niyonsengaeric  please drop tables in the database, then recreateTables. The source of the cause might be that I have changed constraints of a an existing table.

**Comment by Niyonsengaeric** [comment]
> > Screen Shot 2020-02-19 at 15 10 47
> > @Kagorora I followed your PR description but, it stacks on creating the tables!
> 
> @Niyonsengaeric please drop tables in the database, then recreateTables. The source of the cause might be that I have changed constraints of a an existing table.

@Kagorora  the create table is now working

**Comment by Niyonsengaeric** [comment]
please use #id(with the link) on `What are the relevant pivotal tracker stories?`

**Comment by Kagorora** [comment]
> please use id(with the link) on `What are the relevant pivotal tracker stories?`

@Niyonsengaeric I have implemented your feedback

**Review by Niyonsengaeric — APPROVED** [review]

**Review by EugeneMunya — APPROVED** [review]

File: src/controllers/request.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import jwt from 'jsonwebtoken';
+import environment from 'dotenv';
+import models from '../db/models';
+
+environment.config();
+
+export default class usersController {
+
+  static async findAllMyRequest(req, res) {
+    const allMyRequest = await models.Request.findAll({
```
Comment (MuhanguziDavid) [inline src/controllers/request.js:11]: This command seems to be repeated on line 14 of this file

File: src/db/models/request.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+module.exports = (sequelize, DataTypes) => {
+  const Request = sequelize.define('Request', {
+    userEmail: DataTypes.STRING,
```
Comment (MuhanguziDavid) [inline src/db/models/request.js]: This should be a foreign key referencing to a row in the `Users` table

File: src/controllers/request.js (hunk 1 above)
Comment (Kagorora) [reply src/controllers/request.js:11]: @MuhanguziDavid , That was a mistake , I have removed the first one(one outside the try and catch).
       Here is the corrected file
Screen Shot 2020-02-20 at 15 12 48

File: src/db/models/request.js (hunk 2 above)
Comment (Kagorora) [reply src/db/models/request.js]: Here's the implementation, I have made `userEmail` the foreignKey in request table

Screen Shot 2020-02-20 at 15 30 19

File: src/controllers/request.js (hunk 1 above)
Comment (MuhanguziDavid) [reply src/controllers/request.js:11]: Cool. Ensure that your code is properly indented.

**Review by T2Wil — CHANGES_REQUESTED** [review]
Hey @Kagorora ,
Could you change the endpoint name
from : `localhost:4000/api/v1/auth/myRequest`
to : `localhost:4000/api/v1/trips/myRequest`
As this endpoint belongs to trips not authorization.

**Comment by Kagorora** [comment]
> Hey @Kagorora ,
> Could you change the endpoint name
> from : `localhost:4000/api/v1/auth/myRequest`
> to : `localhost:4000/api/v1/trips/myRequest`
> As this endpoint belongs to trips not authorization.

>@william
I have made the changes you suggested, here is a screenshot

Screen Shot 2020-02-21 at 20 27 21

**Review by T2Wil — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -66,6 +66,7 @@ Open terminal from your computer
 
 `- PATCH /auth/logout - Logout a user`
 
+`- GET /auth/myRequest - view all my request history`
```
Comment (MuhanguziDavid) [inline README.md]: change this to `/trips/myRequest`

File: src/tests/request.test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import jwt from 'jsonwebtoken';
+import app from '../app';
+import mockData from './mockData';
+
+chai.use(chaiHttp);
+chai.should();
+
+const userSignUp = () => {
+  describe('View all my Requests.(GET) ', () => {
+    it('it should return 200 on successful signIn', (done) => {
```
Comment (MuhanguziDavid) [inline src/tests/request.test.js:12]: This test seems to be repeated on line 32! Any reason why?

Comment (Kagorora) [reply src/tests/request.test.js:12]: Yes, I repeated that login on view all requests as  I refer to the userId that I get from the token generated when a user logs in to make a search of all his requests. 

 Here is my search for viewing all the requests.

Screen Shot 2020-02-22 at 13 44 46

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Pull request merged**